app/test/block_production_test.go

```diff
@@ -0,0 +1,52 @@
+package app
+
+import (
+	"testing"
+	"time"
+
+	appns "github.com/celestiaorg/celestia-app/pkg/namespace"
+	"github.com/celestiaorg/celestia-app/test/util/testnode"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/stretchr/testify/suite"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+)
+
+func TestBlockProductionestSuite(t *testing.T) {
+	suite.Run(t, new(BlockProductionestSuite))
+}
+
+type BlockProductionestSuite struct {
+	suite.Suite
+
+	accounts []string
+	cctx     testnode.Context
+}
+
+func (s *BlockProductionestSuite) SetupSuite() {
+	if testing.Short() {
+		s.T().Skip("skipping full node integration test in short mode.")
+	}
+	t := s.T()
+
+	accounts := make([]string, 40)
+	for i := 0; i < 40; i++ {
+		accounts[i] = tmrand.Str(10)
+	}
+
+	cfg := testnode.DefaultConfig().
+		WithAccounts(accounts).
+		WithTimeoutCommit(10 * time.Second)
```

Comment:
Just so I understand correctly, if timeout commit is set to 10 seconds, does the first height start 10 seconds after genesis or still at genesis. I'm trying to understand why the node doesn't propose a block instantly before the transaction gets posted.

Reply:
I have made further clarifications in [this comment](https://github.com/celestiaorg/celestia-app/pull/2220/commits/124bb200c7f715479051e078f1d2a93c11f70278) to explicitly state that height 1 has not started yet.

Reply:
Will we want for mainnet that the first height is actually started at `GenesisTime`?

Reply:
Can you please elaborate on your question?

Reply:
Well currently if I set the genesis time to 5pm, the first height actually starts 10 seconds later. Thus the first block is produced perhaps ~12 seconds after genesis. I was wondering whether that is okay or if we'd want the first height to start immediately

Reply:
We don't expect validator nodes to use timeout commit config option in the mainnet (we don't advertise so), we use this option here for testing purposes only. Therefore, in mainnet, the network will start proposing the first block immediately after Genesis time I.e., after loading Genesis state. 

Trusted to your question, Validators can start the first block immediately in the mainnet (we are also going to address #2221 which ensures block 1 would be empty anyway).